src/shared/contexts/dbrps.tsx

```diff
@@ -111,12 +121,16 @@ export const DBRPProvider: FC<Props> = ({children, scope}) => {
       })
   }
 
+  const hasDBRPs = useCallback((): boolean => {
+    return dbrps.length > 0
+  }, [dbrps])
+
   return useMemo(
     () => (
-      <DBRPContext.Provider value={{loading, dbrps}}>
+      <DBRPContext.Provider value={{loading, dbrps, hasDBRPs}}>
         {children}
       </DBRPContext.Provider>
     ),
-    [loading, dbrps]
+    [loading, dbrps, hasDBRPs, children]
```

Comment:
It's a good question! One reason adding `children` in the dependency list is because of the eslint warning. Another reason is that `children` is expected to re-render whenever there is a state update in `children`. For example, in the case here, whenever the script language changes, we want part of the children (schema list) to re-render

Reply:
It's a reasonable question! The previous PR that introduced this file `src/shared/contexts/dbrps.tsx` wasn't used anywhere in the UI, therefore I didn't caught it until this PR 😅